src/reanimated2/platform-specific/webAnimations.ts

```diff
@@ -0,0 +1,278 @@
+    ? getRandomDelay((config as CustomConfig).delayV)
+    : config.delayV / 1000;
+}
+
+export function getDurationFromConfig(
+  config: ConfigType,
+  isLayoutTransition: boolean,
+  animationName: AnimationTypes
+): number {
+  const defaultDuration = isLayoutTransition
+    ? 0.3
+    : Animations[animationName].duration;
+
+  if (!('durationV' in config)) {
+    return defaultDuration;
+  }
+
+  config = config as CustomConfig;
+
+  return config.durationV !== undefined
+    ? config.durationV / 1000
+    : defaultDuration;
+}
+
+export function getCallbackFromConfig(config: ConfigType): (() => void) | null {
+  if (!('callbackV' in config)) {
+    return null;
+  }
+
+  config = config as CustomConfig;
+
+  return config.callbackV !== undefined ? config.callbackV : null;
+}
+
+export function getRandomDelay(maxDelay = 1000): number {
+  return Math.floor(Math.random() * (maxDelay + 1)) / 1000;
+}
+
+export function areDOMRectsEqual(r1: DOMRect, r2: DOMRect): boolean {
+  return JSON.stringify(r1) === JSON.stringify(r2);
+}
+
+export function setElementAnimation(
+  element: HTMLElement,
+  animationConfig: AnimationConfig
+) {
+  const { animationName, duration, delay, easing } = animationConfig;
+
+  element.style.animationName = animationName;
+  element.style.animationDuration = `${duration}s`;
+  element.style.animationDelay = `${delay}s`;
+  element.style.animationTimingFunction = easing;
+  element.style.animationFillMode = 'forwards'; // Prevents returning to base state after animation finishes.
+
+  element.onanimationend = () => animationConfig.callback?.();
+}
+
+export function handleEnteringAnimation(
+  element: HTMLElement,
+  animationConfig: AnimationConfig
+): void {
+  const { delay } = animationConfig;
+
+  // If `delay` === 0, value passed to `setTimeout` will be 0. However, `setTimeout` executes after given amount of time, not exactly after that time
+  // Because of that, we have to immediately toggle on the component when the delay is 0.
+  if (delay === 0) {
+    element.style.visibility = 'initial';
+  } else {
+    setTimeout(() => {
+      element.style.visibility = 'initial';
+    }, delay * 1000);
+  }
+
+  setElementAnimation(element, animationConfig);
+}
+
+export function handleExitingAnimation(
+  element: HTMLElement,
+  animationConfig: AnimationConfig
+): void {
+  const parent = element.offsetParent;
+  const tmpElement = element.cloneNode() as HTMLElement;
+
+  // After cloning the element, we want to move all children from original element to its clone. This is because original element
+  // will be unmounted, therefore when this code executes in child component, parent will be either empty or removed soon.
+  // Using element.cloneNode(true) doesn't solve the problem, because it creates copy of children and we won't be able to set their animations
+  //
+  // This loop works because appendChild() moves element into its new parent instead of copying it and then inserting copy into new parent
```

Comment:
Last part of this comment is unclear. It moves the element but also it inserts a copy?

Reply:
Done in [7f8735d](https://github.com/software-mansion/react-native-reanimated/pull/4805/commits/7f8735deb90c38024a96cbe8a49260758ee8097f)